### ISSUE TITLE
doc: document the required naming for models' constraints

### ISSUE THREAD

**Issue by nharraud** [inveniosoftware/invenio-db#78]
**Problem:**
Since #65 It is necessary in some cases to explicitly name constraints as explained in #69. We need to document that. We should also say that the name should be less than 64 characters (see [postgresql](https://www.postgresql.org/docs/9.1/static/sql-syntax-lexical.html) and [mysql](https://dev.mysql.com/doc/refman/5.7/en/identifiers.html) limitations) and that it should be as close as possible to the naming convention defined in [`shared.py`](https://github.com/inveniosoftware/invenio-db/blob/master/invenio_db/shared.py)

Note: the naming is needed both for models and alembic scripts.

Also the tests should be made with mysql. Postgresql seems to truncate the name by default so everything seems fine but it isn't.

**Comment by jirikuncar** [comment]
@nharraud we can update the default naming scheme with function/class that check the length of the resulting name and raise and exception directly.

```python
In [1]: class str_mod_limit(str):
   ...:
   ...:     def __mod__(self, *args, **kwargs):
   ...:         result = super(str_mod_limit, self).__mod__(*args, **kwargs)
   ...:         if len(result) > 64:
   ...:             raise RuntimeError('String limit reached')
   ...:         return result
```

**Comment by nharraud** [comment]
@jirikuncar this would be a nice to have yes 👍 

Just as a side note for the future implementation, the code should be `if len(result) > 63` as it is "less than 64".

However I would do it after releasing invenio-db as it is. More and more PRs are being blocked because of invenio-db not being released (example: secret_key update feature), and we can't release it because some modules are still breaking the naming convention, which means users would not be able to test invenio-3/invenio-app-ils. I just finished investigating what needs to be done for those modules.

**Comment by lnielsen** [comment]
Closed by #97
